src/frontend/js/pages/TeacherDashboardContractsLayout/components/ContractFiltersBar/index.tsx

```diff
@@ -42,9 +42,15 @@ const ContractFiltersBar = ({
   hideFilterOrganization = false,
   hideFilterSignatureState = false,
 }: ContractFiltersBarProps) => {
+  const {
+    items: organizations,
+    states: { isFetched: isOrganizationsFetched },
+  } = useOrganizations(undefined, { enabled: !hideFilterOrganization });
+
+  const hasMultipleOrganizations = isOrganizationsFetched && organizations.length > 1;
   return (
     <div className="dashboard__page__actions-row dashboard__page__actions-row--end">
-      {!hideFilterOrganization && (
+      {!hideFilterOrganization && hasMultipleOrganizations && (
         <OrganizationContractFilter
           defaultValue={defaultValues?.organization_id}
```

Comment:
I wonder if this is the right place to do that ?
Maybe the parent component should process the `hasMultipleOrganizations` then set `hideFilterOrganization` accordingly 🤔

Reply:
I see.
Could it be the other way with `ContractFiltersBar` looking up for route params then removing this `hideOrganizationFilter` prop ?
I like the idea of `ContractFiltersBar` component to handle the responsibility to display or not it's parts.

Reply:
From my point of view, this is weird as there is props to show hide filter but according the context those props can have no effects. IMO it could be confusing.